api/src/service/RpcClient.ts

```diff
@@ -101,22 +101,97 @@ export class RpcClient {
     });
   }
 
-  public invoke(method: string, ...params: any[]): any {
+  private generateRequest = (
+    stream: String,
+    keys: String | String[],
+    streamitem: ItemToPublish,
+    address?: String,
+    offchain?: Boolean,
+  ): Result.Type<any> => {
+    let params = [stream, keys, streamitem];
+    let method: string;
+    if (config.signingMethod === "user" && address) {
+      method = "publishfrom";
+      params = offchain ? [address, ...params, "offchain"] : [address, ...params];
+    } else if (config.signingMethod === "node" || (config.signingMethod === "user" && !address)) {
+      method = "publish";
+      params = offchain ? [...params, "offchain"] : params;
+    } else {
+      logger.error(`Failed to publish event. Invalid signing method: ${config.signingMethod}`);
+      return new RpcError(
+        500,
+        `Failed to publish data to the chain. Invalid signing method: ${config.signingMethod}`,
+        {},
+        "",
+      );
+    }
+    return { method, params };
```

Comment:
can this be typed generateRequest : Request
```
Request= {
  method: string
  params: any
}
```

Reply:
I used the already existing RpcRequest, which I updated because it was not up to date (we don't use a request identifier, params are not optional)